DigitalOffice/Shared/Kernel/AccessValidator/AccessValidator.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using MassTransit;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Options;
+using LT.DigitalOffice.Kernel.Broker;
+using LT.DigitalOffice.Kernel.AccessValidator.Requests;
+using LT.DigitalOffice.Kernel.AccessValidator.Interfaces;
+
+namespace LT.DigitalOffice.Kernel.AccessValidator
+{
+    public class AccessValidator : IAccessValidator
+    {
+        private readonly IBus bus;
+        private Guid userId;
+        private readonly RabbitMQOptions options;
+        private readonly IHttpContextAccessor httpContextAccessor;
+
+        public AccessValidator(IHttpContextAccessor httpContextAccessor, IBus bus, IOptions<RabbitMQOptions> options)
+        {
+            this.httpContextAccessor = httpContextAccessor;
+            this.options = options.Value;
+            this.bus = bus;
+        }
+
+        private Guid GetCurrentUserId()
+        {
+            return Guid.Parse(httpContextAccessor.HttpContext.User.FindFirst(ClaimTypes.NameIdentifier).Value);
```

Comment:
Could you please explain how to send a userId via postman and receive userId here?

Reply:
@ifelseelif well, I can't tell you how to test it via Postman, because I haven't done that yet.

I know for sure that this command works when you log in the web application via Identity and UserId must be stored in a cookie.

As for Postman, you can look up how to pass ASP Net Core authorization in Postman. I can't tell you much about the topic. I'm still waiting for the approval that I can implement Access Validator like this.

Reply:
Implementation has changed to Headers. So if you'd want to test it via Postman, use them.

Reply:
![image](https://user-images.githubusercontent.com/4159835/91506562-1241b780-e8db-11ea-8529-2533f90717b0.png)